### PR TITLE
Added maintenance endpoints 

### PR DESCRIPTION
**Pull request by roiaharon-pepperi**

### CODE REVIEW

File: endpoints/maintenance.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    private resourceName: ResourceName = 'all_activities';
+    private service: PapiClient;
+
+    constructor(service: PapiClient) {
+        this.service = service;
+    }
+
+    archived = {
+        type: (type: ResourceName) => {
+            return new Endpoint<ArchivedObject>(this.service, `/maintenance/archived/${type}`);
```
Comment (Chasky-H) [inline endpoints/maintenance.ts:15]: This only supports `iter` not the other functions on Endpoint

File: endpoints/maintenance.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        },
+        job_info: async (uuid: string): Promise<MaintenanceJobInfo> => {
```
Comment (Chasky-H) [inline endpoints/maintenance.ts:17]: jobInfo

File: endpoints/maintenance.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        const url = this.GetMaintenanceApiUrl(params, 'archive');
+        return await this.service.post(url, body);
+    }
+
+    async unarchive(params: any = {}, body: any = undefined): Promise<MaintenanceJobResult> {
+        const url = this.GetMaintenanceApiUrl(params, 'unarchive');
+        return await this.service.post(url, body);
+    }
+
+    private GetMaintenanceApiUrl(params: any, funcName: string) {
```
Comment (Chasky-H) [inline endpoints/maintenance.ts:37]: getMaintenanceApiUrl

File: endpoints/maintenance.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        return this;
+    }
+
+    async archive(params: any = {}, body: any = undefined): Promise<MaintenanceJobResult> {
+        const url = this.GetMaintenanceApiUrl(params, 'archive');
+        return await this.service.post(url, body);
+    }
+
+    async unarchive(params: any = {}, body: any = undefined): Promise<MaintenanceJobResult> {
+        const url = this.GetMaintenanceApiUrl(params, 'unarchive');
```
Comment (Chasky-H) [inline endpoints/maintenance.ts:33]: isn't a body mandatory?
what is the body format?
Maybe create an Interface.
Same for `archive`.

File: entities/maintenance.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        ExternalID?: string;
+    };
+    Type?: {
+        ID?: number;
+        Name?: string;
+    };
+    Status?: {
+        ID?: number;
+    };
+}
```
Comment (Chasky-H) [inline entities/maintenance.ts:36]: None of these fields should be marked as optional, they all arrive from the API, Correct?

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "@pepperi-addons/papi-sdk",
-    "version": "1.0.34",
+    "version": "1.0.35",
```
Comment (Chasky-H) [inline package.json:3]: You are adding functionality, so you should bump a minor.
See the README

**Comment by roiaharon-pepperi** [comment]
@Chasky-H  made the requested you've asked. please review

File: endpoints/maintenance.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        return this;
+    }
+
+    async archive(body: ArchiveBody): Promise<MaintenanceJobResult> {
+        const url = this.getMaintenanceApiUrl('', 'archive');
+        return await this.service.post(url, body);
+    }
+
+    async unArchive(params: any = {}, body: UnArchiveBody): Promise<MaintenanceJobResult> {
+        const url = this.getMaintenanceApiUrl(params, 'unarchive');
```
Comment (Chasky-H) [inline endpoints/maintenance.ts:40]: Are there any params? 
Why have this parameter at all?

File: entities/maintenance.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+export interface MaintenanceJobInfo {
+    UUID: string;
+    Success: boolean;
+    RecordsAffected: number;
+    ErrorMessage: string;
+}
+
+export interface UnArchiveBody {
+    object: {
```
Comment (Chasky-H) [inline entities/maintenance.ts:14]: Are you sure this is what the unArchive body looks like?

File: endpoints/maintenance.ts (hunk 7 above)
Comment (roiaharon-pepperi) [reply endpoints/maintenance.ts:40]: copy + paste mistake. removed it

File: entities/maintenance.ts (hunk 8 above)
Comment (roiaharon-pepperi) [reply entities/maintenance.ts:14]: yes, of course. take a look at the documentation:
https://apidesign.pepperi.com/archive/bulk-async-unarchive-of-data-resources

**Comment by roiaharon-pepperi** [comment]
@Chasky-H merged master into this brach and resolved all issues

**Pull request merged**